cscs-checks/libraries/math/dgemm.py

```diff
@@ -2,49 +2,87 @@
 import reframe.utility.sanity as sn
 
 
+@rfm.required_version('>=2.14')
+@rfm.simple_test
 class DGEMMTest(rfm.RegressionTest):
     def __init__(self):
         super().__init__()
         self.descr = 'DGEMM performance test'
         self.sourcepath = 'dgemm.c'
-        self.executable_opts = ['5000', '5000', '5000']
-        self.sanity_patterns = sn.assert_found(
-            r'Time for \d+ DGEMM operations', self.stdout)
-        self.maintainers = ['AJ']
-        self.tags = {'production'}
 
+        self.sanity_patterns = self.eval_sanity()
+        # the perf patterns are automaticaly generated inside sanity
+        self.perf_patterns = {}
```

Comment:
I would say to either create another function for the `perf_patterns` or put a flag argument in the `eval_sanity`  to distinguish between sanity/performance checking.

Reply:
@teojgo, isn't it practical and elegant to generate the performance patterns inside the sanity?
I really liked the approach implemented here. It was also the same approach done on the COSMO tests.

Reply:
Yes, I would just create another function to separate sanity and performance instead of setting the `perf_patterns` to an empty dict and then assign it inside the `eval_sanity`,

Reply:
@teojgo The problem is that the `perf_patterns` that @victorusu is generating here depend on the actual hostname information; there is a separate key per node. I don't know if this is possible to be done in a different way. Even if we had feature #565 in place.